### ISSUE TITLE
Rewrite rules with interval arguments are not supported

### ISSUE THREAD

**Issue by Zinfour** [agda/agda#4384]
With this code:
```agda
{-# OPTIONS --cubical --rewriting #-}
open import Agda.Builtin.Cubical.Path
open import Agda.Primitive.Cubical

data A : Set where
  a  : A
  eq : a ≡ a
  
bad : ∀ i → eq i ≡ a
bad i j = eq (primIMin i (primINeg j))

{-# BUILTIN REWRITE _≡_ #-}
{-# REWRITE bad #-}
```
I get the following error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/TypeChecking/Rewriting/NonLinPattern.hs:65
```
My agda version is `2.6.1-475c7e5`.

**Comment by jespercockx** [comment]
Here's the offending line: https://github.com/agda/agda/blob/475c7e515e2edbcff789f4fa3836b64828e1e8dc/src/full/Agda/TypeChecking/Rewriting/NonLinPattern.hs#L65

It does not really surprise me, as the implementor of rewrite rules (me) knows little of the implementation of cubical, and I imagine @Saizan doesn't know all the details the implementation of rewrite rules either. I could try to make a guess of how rewrite rules with interval variables should be implemented, but without thinking about it for a long time I would probably mess things up. Maybe the best thing to do right now would be to add a proper error message that rewrite rules with interval variables are not yet supported, and wait until someone actually knows what should be done.

**Comment by jespercockx** [comment]
(Adding to the 2.6.1 milestone for replacing the internal error with a proper error message, actually making this work will have to wait until later.)

**Comment by Saizan** [comment]
From reading the documentation on  `PatternFrom` I would be inclined to really mirror the case for `Apply`, except for swapping `Pi` with the `PathP` def.

**Comment by jespercockx** [comment]
I guess that would make sense. But then one would also have to update the matching algorithm (in `NonLinMatch`) to deal with `IApply` patterns, and other places as well where I just put `__IMPOSSIBLE__` for now.

**Comment by andreasabel** [comment]
@Saizan : What is the status of this issue?  Could you give a summary of what you did and what remains to be done?

**Comment by Saizan** [comment]
@andreasabel I implemented what me and @jespercockx discussed above, not sure if there's more to be done though.

Would probably be good to add a few more test cases, e.g. for higher order patterns.

**Comment by andreasabel** [comment]
Thanks @Saizan !
Patch e2eb867be0353cec5ad1a3edf0ef209b3ed6aef4 does not pick cleanly onto `maint-2.6.2`, so this fix is not scheduled for 2.6.2.1.
